audmetric/core/api.py

```diff
@@ -237,6 +341,118 @@ def edit_distance(
+        typing.Union[bool, int, float],
+        typing.Sequence[typing.Union[bool, int, float]]
+    ],
+    prediction: typing.Union[
+        typing.Union[bool, int, float],
+        typing.Sequence[typing.Union[bool, int, float]]
+    ],
+) -> typing.Tuple[float, namedtuple]:
+    r"""Equal error rate for verification tasks.
+
+    The equal error rate (EER) is the point
+    where false non-match rate (FNMR)
+    and the impostors or false match rate (FMR)
+    are identical.
+    The FNMR indicates
+    how often an enrolled speaker was missed.
+    The FMR indicates
+    how often an impostor was verified as the enrolled speaker.
+
+    In practice the score distribution is not continuous
+    and an interval is returned instead.
+    The EER value will be set as the midpoint
+    of this interval::footcite:`Maio2002`
+
+    .. math::
+
+        \text{EER} = \frac{
+            \min(\text{FNMR}[t], \text{FMR}[t])
+            + \max(\text{FNMR}[t], \text{FMR}[t])
+        }{2}
+
+    with :math:`t = \text{argmin}(|\text{FNMR} - \text{FMR}|)`.
+
+    ``truth`` may only contain entries like ``[1, 0, True, False...]``,
+    whereas prediction values
+    can also contain similarity scores, e.g. ``[0.8, 0.1, ...]``.
+
+    The implementation is identical with the one provided
+    by the pyeer_ package.
+
+    .. footbibliography::
+
+    .. _pyeer: https://github.com/manuelaguadomtz/pyeer
+
+    Args:
+        truth: ground truth classes
+        prediction: predicted classes
+
+    Returns:
+        * equal error rate (EER)
+        * namedtuple containing
+          ``fmr``,
+          ``fnmr``,
+          ``thresholds``,
+          ``threshold``
+          whereas the last one corresponds to the threshold
+          corresponding to the returned EER
+
+    Raises:
+        ValueError: if ``truth`` contains values
+            different from ``1, 0, True, False``
+
+    Example:
+        >>> truth = [0, 1, 0, 1, 0]
+        >>> prediction = [0.2, 0.8, 0.4, 0.5, 0.5]
+        >>> eer, stats = equal_error_rate(truth, prediction)
+        >>> eer
+        0.16666666666666666
+        >>> stats.threshold
+        0.5
+
+    """
+    Stats = namedtuple(
+        'stats',
+        [
+            'fmr',  # False match rates (FMR)
+            'fnmr',  # False non-match rates (FNMR)
+            'thresholds',  # Thresholds
+            'threshold',  # verification threshold for EER
+        ],
+    )
+    fmr, fnmr, thresholds = detection_error_tradeoff(truth, prediction)
+    diff = fmr - fnmr
+    # t1 and t2 are our time indices
+    t2 = np.where(diff <= 0)[0]
+    if len(t2) > 0:
+        t2 = t2[0]
+    else:
+        warnings.warn(
+            'The false match rate '
+            'and false non-match rate curves '
+            'do not intersect each other.',
+            RuntimeWarning,
+        )
+        eer = 1.0
+        threshold = float(thresholds[0])
+        return eer, Stats(fmr, fnmr, thresholds, threshold)
```

Comment:
In `pyeer` ( https://github.com/manuelaguadomtz/pyeer/blob/master/pyeer/eer_stats.py#L310 ) constants are returned as statistics. I have seen that you tested `eer` and `threshold` for this case. Will the computed `threshold` always be `0.0` as in `pyeer`? I just wanted to make sure.

Reply:
What they return there in `pyeer` is `index for EERlow and EERhigh, EERlow, EERhigh, EER`, so not the threshold but the first return value is the index that corresponds to the chosen EER value, so we could use that index to get our threshold.

Which means `0` in there code becomes `thresholds[0]` in our code.

Reply:
Thanks for clarification